src/DotLiquid.Tests/ContextTests.cs

```diff
@@ -785,34 +790,78 @@ public void TestNestedDynamicAsVariable()
             Assert.AreEqual("Hello", _context["dynamic.lambda.name"]);
         }
 
+        private class ExpandoModel
+        {
+            public int IntProp { get; set; }
+            public string StrProp { get; set; }
+            public ExpandoObject Props { get; set; }
+        }
+
+        // Test case for [Issue #350](https://github.com/dotliquid/dotliquid/issues/350)
+        [Test]
+        public void TestNestedExpandoTemplate_Issue350()
+        {
+            var model = new ExpandoModel()
+            {
+                IntProp = 23,
+                StrProp = "from string prop",
+                Props = new ExpandoObject()
+            };
+            var dic = (IDictionary<string, object>)model.Props;
+            dic.Add("Key1", "from expando prop");
+
+            Template.RegisterSafeType(typeof(ExpandoModel), new[] { "IntProp", "StrProp", "Props" });
+            const string templateString = @"Int: '{{IntProp}}'; String: '{{StrProp}}'; Expando: '{{Props.Key1}}'";
+            var t = Template.Parse(templateString);

```

Comment:
Same for single letter variables. Just don't use them. Always use a word from the English dictionary that describes the purpose of the variable.

Reply:
Note that there might be older code that uses ambiguous names. Don't copy that :-)